src/actions/vstudio/vs2005_csproj.lua

```diff
@@ -314,8 +330,44 @@
 				_x(2,'<Reference Include="%s" />', name)
 			end
 		end)
+	end
 
-		_p(1,'</ItemGroup>')
+
+--
+-- Write the list of NuGet references.
+--
+
+	function cs2005.nuGetReferences(prj)
+		if _ACTION >= "vs2010" then
+			for i = 1, #prj.nuget do
+				local package = prj.nuget[i]
+				_x(2, '<Reference Include="%s">', vstudio.nuget2010.packageId(package))
+
+				-- We need to write HintPaths for all supported framework
+				-- versions. The last HintPath will override any previous
+				-- HintPaths (if the condition is met that is).
+
+				for _, frameworkVersion in ipairs(cs2005.identifyFrameworkVersions(prj)) do
+					local assembly = vstudio.path(
+						prj,
+						p.filename(
+							prj.solution,
+							string.format(
+								"packages\\%s\\lib\\%s\\%s.dll",
+								vstudio.nuget2010.packageName(package),
+								cs2005.formatNuGetFrameworkVersion(frameworkVersion),
+								vstudio.nuget2010.packageId(package)
+							)
+						)
+					)
+
+					_x(3, '<HintPath Condition="Exists(\'%s\')">%s</HintPath>', assembly, assembly)
+				end
+
+				_p(3, '<Private>True</Private>')
```

Comment:
One thing to note about this which we've had to fix on our internal version - this doesn't respect NoCopyLocal. On our version we replaced this line with:
local cfg = project.getfirstconfig(prj)
if premake.config.isCopyLocal(prj, package, cfg) then
      _p(3, '<Private>True</Private>')
 else
       _p(3, '<Private>False</Private>')
end